newrelic/hooks/database_psycopg2.py

```diff
@@ -63,6 +63,10 @@ class ConnectionSaveParamsWrapper(DBAPI2ConnectionWrapper):
 
     def __enter__(self):
         transaction = current_transaction()
+        if not transaction:
```

Comment:
It's surprising to me that this has been wrong for so long but I don't actually see in the existing code where it is gated by this check higher up so this seems correct to me.